### DISCUSSION TITLE
cfg.sd_active = true;

### DISCUSSION THREAD

**Discussion by mydroidbox** [pschatzmann/arduino-audio-tools#651] [Q&A] [answered]
void wifiConnected(){
  
  kit.begin(cfg);
  player.setVolume(1.0);
  player.begin();
  player.setPath("/wCon.mp3"); 
}

Hello Phil.
I'm calling a sound sample from spiffs like above
running your player-sd-audiokit sketch, modified to play from spiffs instead of sd card.  I'm also running an ftp server which accesses the audiiokit's sd card.
Nothing is run at boot, both the server and sound samples from spiffs are called in functions when needed. I can call sound samples from spiffs before connecting to the server, however if I call a sound sample after the server is connected, your script even though it is reading from spiffs
and not from sd sets cfg.sd_active = true; preventing the server from accessing sd card. I tried setting cfg.sd_active = false;  in setup as well as #define AUDIOKIT_SETUP_SD 0 in audioKitSettings with no luck
let me know what else I could try. Below is debug when server tries to access sd.
Thanks.

[ 28501][W][sd_diskio.cpp:174] sdCommand(): no token received
[ 28601][W][sd_diskio.cpp:174] sdCommand(): no token received
[ 28701][W][sd_diskio.cpp:174] sdCommand(): no token received
[ 28801][E][sd_diskio.cpp:199] sdCommand(): Card Failed! cmd: 0x0d
[ 28801][E][sd_diskio.cpp:621] ff_sd_status(): Check status failed
[ 28803][W][sd_diskio.cpp:174] sdCommand(): no token received
[ 28908][W][sd_diskio.cpp:174] sdCommand(): no token received
[ 29008][W][sd_diskio.cpp:174] sdCommand(): no token received
[ 29107][E][sd_diskio.cpp:199] sdCommand(): Card Failed! cmd: 0x00
[ 29107][W][sd_diskio.cpp:516] ff_sd_initialize(): GO_IDLE_STATE failed
[ 29108][E][vfs_api.cpp:293] VFSFileImpl(): opendir(/sd/) failed
[ 32899][D][WiFiClient.cpp:546] connected(): Disconnected: RES: 0, ERR: 128
Connection closed!

**Comment by pschatzmann** [comment, marked as answer]
I assume you are using the AudioKit board ?
Please read the wiki in the AudioKit project: It should have all the information!

Did you try the https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-audiokit/player-sdmmc-audiokit/player-sdmmc-audiokit.ino example ?

**Reply by mydroidbox** [reply]
Yes I'm using the audiokit . As mentioned in a previous post my board is modified to use 
SPI.begin(14, 2, 15, 18);  Pin 13 is now used for touch. 
This SPI is already defined in setup and works with all sd sketches.
Not seeing anything that i missed in the wiki.
Getting below for sdmmc:

E (60754) sdmmc_common: sdmmc_init_ocr: send_op_cond (1) returned 0x107
E (60755) vfs_fat_sdmmc: sdmmc_card_init failed (0x107).
[ 60758][E][SD_MMC.cpp:138] begin(): Failed to initialize the card (0x107). Make sure SD card lines have pull-up resistors in place.
[E] AudioSourceSDMMC.h : 47 - SD_MMC.begin failed

**Reply by pschatzmann** [reply]
Did you try to google which pins are used by SDMMC: if I remember right pint 13 is MicroSD D3,  
I am avoiding SDMMC because it is also conflicting with some button pins...

**Reply by mydroidbox** [reply]
I just looked at the audioSource header file. That might me a little tricky to implement with my setup. Is there any suggestion you could give for making the first concept work. Spiffs for audio samples and sd for server.

**Reply by pschatzmann** [reply]
> Did you try to google which pins are used by SDMMC: if I remember right pint 13 is MicroSD D3, I am avoiding SDMMC because it is also conflicting with some button pins...

The ESP32 provides a SDMMC example which describes the used pins. 
Did you try this ?

**Reply by pschatzmann** [reply]
I dont understand what you want to achieve: To mix SPIFFS and SMMC does not make any sense to me...

**Reply by mydroidbox** [reply]
I don't want to use sdmmc at all. I merely want to use sd for my ftp server and to call samples from spiffs without sd_active becoming true and interfering with server accessing sd card.

**Reply by mydroidbox** [reply]
To simplify my question. How can  I run my function calling sound samples from spiffs without sd_active becoming true.

**Reply by pschatzmann** [reply]
Oh, I must have mixed up something...

**Reply by pschatzmann** [reply]
> To simplify my question. How can I run my function calling sound samples from spiffs without sd_active becoming true.

I never tried, but I would expect so: Why don't you just use the [spiffs example](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-player/player-spiffs-i2s/player-spiffs-i2s.ino) ? spiffs does not care if you set the sd active or inactive in the config.

**Reply by mydroidbox** [reply]
I am in essence running the spiffs example. Unless I'm wrong the only difference between the 2 is audiosource and that one outputs to i2s while the other ouputs to kit.

**Reply by pschatzmann** [reply]
ps. The relevant documentation is in the Wiki about the player: https://github.com/pschatzmann/arduino-audio-tools/wiki/The-Audio-Player-Class

**Reply by mydroidbox** [reply]
I looked at your suggested links.
I'm a beginner coder. While I appreciate your reluctance to spoonfeed. I only ask for help when it  is beyond me. Let me know if you can help. Thanks.
